graph/src/util/error.rs

```diff
@@ -17,3 +17,10 @@ macro_rules! ensure {
         }
     };
 }
+
+#[macro_export]
+macro_rules! bail {
+    ($($err:tt)*) => {
+        return Err(anyhow::anyhow!($($err)*).into());
+    };
+}
```

Comment:
Why don't we just use the existing `anyhow::bail!` macro? If this is truly needed, maybe add a comment why.

Reply:
It doesn't call `.into()`, yhea needs a comment.